java-logging-appinsights/src/main/java/uk/gov/hmcts/reform/logging/appinsights/telemetry/initializers/SpringApplicationName.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.hmcts.reform.logging.appinsights.telemetry.initializers;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is a hack we need until Microsoft delivers proper support for Spring applications. Telemetry initializers
+ * defined in ApplicationInsights.xml are not instrumented by Spring, we need to expose the application name via
+ * a static method in order to access it.
+ */
+@Component
+public class SpringApplicationName {
+    private static String value;
+
+    public static String get() {
+        return value;
+    }
+
+    @Value("${spring.application.name}")
+    public void setValue(String value) {
+        SpringApplicationName.value = value;
+    }
+}
```

Comment:
Maybe worth just using Microsoft's magic instead of Spring Boot since it is related to AppInsights setup? So instead of creating yet another bean you pass the param field in config:

```xml
<Add type=...>
    <Param name="appName" value="..." />
</Add>
```

Then get rid of this class and add an initialiser constructor. Would make feature free of Spring related code. Probably would make more confusing by trying to pass the app name here. If so, can ignore the idea